dronecan/protocol/7.NodeCapabilities.uavcan

```diff
@@ -0,0 +1,59 @@
+# NodeCapabilities messages
+
+# this message contains information about the capabilities of the node sending the messgage,
+# or if the authoritative flag is set, then it announces the mininum capabilities of all nodes on the bus
+
+# a node should not activate a capability until it has seen an
+# authoritative announcement that the mininum capability on the bus
+# includes the corresponding capability. For example, if a node is
+# capable of doing 8MBit FDCAN then it should set the
+# FLAG_FDCAN_4M_SUPPORT in its announcements, but not start using 4M
+# FDCAN until it has seen an authoritative message with the
+# FLAG_FDCAN_4M_SUPPORT bit set in flags.
+
+# on receipt of this message all nodes should set their active
+# capabilities to the minimum of the received message capabilities and
+# their current capabilities. For example, this means if a node is
+# currently using FDCAN 4M and it sees a packet (whether authoritative
+# or not) without the FDCAN 4M flag set then it should stop using
+# FDCAN and fall back to the next lowest level (in the case of 4M
+# FDCAN this would be 1M bxCAN)
+
+# the central server (usually the DNA server) collates the
+# capabilities of all nodes on the bus and is responsible for
+# announcing the minimum capabilities with the authoritative flag
+# set. The central server will also send an announcement with its own
+# capabilities, which may be higher than the minimum on the bus. This
+# announcement of its own capabilities should not have the
+# authoritative flag set.
+
+# the central server may have a (possibibly configurable) delay
+# between when an increase in minimum network capability happens and
+# when it announces the upgraded capability. A delay on startup is a
+# good idea to allow time for legacy nodes to startup
+
+uint32 FLAG_AUTHORITATIVE         = 1
+# the PUBLISHER_ONLY bit should be set if this node only publishes data, most sensor
+# nodes should set this bit
+uint32 FLAG_PUBLISHER_ONLY        = 2
```

Comment:
What's the purpose of FLAG_PUBLISHER_ONLY? Does it mean literally only publishes and doesn't subscribe to anything or is there some nuance (parameters, etc)?

Reply:
the idea is to distinguish between a node publishing sensor data and nodes that consume sensor data. The reason this matters is for thinks like the old GNSS Fix message. We'd like to stop sending that message to save bandwidth, but we can't do that unless we know that all nodes that consume GNSS messages understand Fix2. We don't care if (for example) a baro/compass sensor node is at the level that understands Fix2, as it doesn't consume any GNSS messages, but we do care if all the consumer nodes (flight controllers, loggers, cameras etc) understand Fix2.
So this is a deliberately broad category to classify nodes as either "publish only" or "publish and consume". That reflects the reality of how DroneCAN networks are actually used, and will allow us to push out replacement messages much more easily.
Note that while I'm using GNSS Fix/Fix2 as an example, the same principle applies to other messages, and the same distinction between nodes can be used